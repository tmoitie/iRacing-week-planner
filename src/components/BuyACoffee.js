// @flow

import axios from 'axios';
import * as React from 'react';

import { useTranslation } from 'react-i18next';
import styles from './styles/buyACoffee.module.css';

export default function BuyACoffee(): React.Node {
  const { t, i18n } = useTranslation();

  const clickBuyCoffee = () => {
    window.dataLayer.push({
      event: 'select_content',
      contentType: 'sponsor',
      contentID: 'Buy A Coffee',
    });

    window.location.assign('https://www.buymeacoffee.com/tmoitie');
  };

  return (
    <button type="button" className={styles.bmcButton} onClick={clickBuyCoffee} onKeyPress={clickBuyCoffee}>
      <img
        src="https://cdn.buymeacoffee.com/buttons/bmc-new-btn-logo.svg"
        alt={t('Buy me a coffee')}
      />
      <span style={{ marginLeft: '5px' }}>
        {t('Buy me a coffee')}
      </span>
    </button>
  );
}
