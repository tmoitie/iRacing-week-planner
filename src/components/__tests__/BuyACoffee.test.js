import React from 'react';
import { create, act } from 'react-test-renderer';
import { describe, test } from '@jest/globals';
import axios from 'axios';
import { useTranslation } from 'react-i18next';

import BuyACoffee from '../BuyACoffee';

describe('components/BuyACoffee', () => {
  test('opens link', async () => {
    const component = create(<BuyACoffee />);

    expect(component.toJSON()).toMatchSnapshot();

    await act(async () => {
      await component.root.findByType('button').props.onClick();
    });

    expect(window.location.assign).toHaveBeenCalledWith('https://www.buymeacoffee.com/tmoitie');
  });
});
