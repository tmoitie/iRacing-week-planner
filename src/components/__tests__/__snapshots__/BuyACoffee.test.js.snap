// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/BuyACoffee opens link 1`] = `
<button
  className="bmcButton"
  onClick={[Function]}
  onKeyPress={[Function]}
  type="button"
>
  <img
    alt="Buy me a coffee"
    src="https://cdn.buymeacoffee.com/buttons/bmc-new-btn-logo.svg"
  />
  <span
    style={
      {
        "marginLeft": "5px",
      }
    }
  >
    Buy me a coffee
  </span>
</button>
`;
